hours/viewsets.py

```diff
@@ -568,43 +568,51 @@ def copy_date_periods(self, request, pk=None):
             if resource_id.strip()
         ]
 
-        target_resources = []
+        filters = map(get_resource_pk_filter, target_resource_ids)
+        q_objects = [Q(**filter) for filter in filters]
+        query_q = Q()
+        for q in q_objects:
+            query_q |= q
+
+        target_resources = (
+            Resource.objects.prefetch_related(
+                "origins__data_source", "parents", "organization"
+            )
+            .filter(query_q)
+            .all()
+        )
+
+        if target_resources.count() != len(target_resource_ids):
+            for target_resource_id in target_resource_ids:
+                try:
+                    Resource.objects.get(**get_resource_pk_filter(target_resource_id))
```

Comment:
This is something I would have liked to do so that we would not need to do another select for getting the non-found resources.